llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -1375,14 +1375,47 @@ InstructionCost RISCVTTIImpl::getCmpSelInstrCost(unsigned Opcode, Type *ValTy,
                           LT.second, CostKind);
   }
 
-  if ((Opcode == Instruction::ICmp || Opcode == Instruction::FCmp) &&
-      ValTy->isVectorTy()) {
-    std::pair<InstructionCost, MVT> LT = getTypeLegalizationCost(ValTy);
-
-    // Support natively.
-    if (CmpInst::isIntPredicate(VecPred))
-      return LT.first * 1;
+  if ((Opcode == Instruction::ICmp) && ValTy->isVectorTy()) {
+    unsigned RVVOp;
+    switch (VecPred) {
+    case CmpInst::ICMP_EQ:
+      RVVOp = RISCV::VMSEQ_VV;
+      break;
+    case CmpInst::ICMP_NE:
+      RVVOp = RISCV::VMSNE_VV;
+      break;
+    case CmpInst::ICMP_UGT:
+      RVVOp = RISCV::VMSLTU_VV;
```

Comment:
Should this be `RISCV::VMSLEU_VV` though if we're swapping the operands? Or is it just a placeholder for a similar cost

Reply:
VMSLTU is correct for swapping operand. UGT should false for equality just like VMSLTU would. VMSLEU would return true for equality.